src/job-exporter/src/main.py

```diff
@@ -54,7 +55,31 @@ def collect(self):
 
 def config_environ():
     """ since job-exporter needs to call nvidia-smi, we need to change
-    LD_LIBRARY_PATH to correct value """
+    LD_LIBRARY_PATH and PATH to correct value """
+    deploy_env = os.environ.get("DEPLOY_ENV")
+    # Refer to issue: https://github.com/Azure/AKS/issues/1271, we need to set nvdia-smi manually
+    if deploy_env == "aks":
+        host_usr_bin_dir = os.environ.get("HOST_USR_BIN_DIR")
+        host_nvidia_bin_dir = os.environ.get("HOST_NVIDIA_BIN_DIR")
+        if not host_usr_bin_dir or not host_nvidia_bin_dir:
+            logger.error("Failed to get HOST_USR_BIN_DIR or HOST_NVIDIA_BIN_DIR env")
+            raise Exception("Environment not set correctly, HOST_NVIDIA_BIN_DIR is{},\
+                             HOST_NVIDIA_BIN_DIR is {}".format(host_usr_bin_dir, host_nvidia_bin_dir))
+
+        if os.path.isfile(host_usr_bin_dir + "/nvidia-smi"):
```

Comment:
use `os.path.join(_dir, 'nvidia-smi')`

Reply:
Fixed

---
In reply to: [336347774](https://github.com/microsoft/pai/pull/3749#discussion_r336347774) [](ancestors = 336347774)